docs/rules/rename-default-import.md

```diff
@@ -0,0 +1,63 @@
+# import/rename-default-import
+
+This rule will enforce a specific binding name for a default package import.
+Works for ES6 imports and CJS require.
+
+
+## Rule Details
+
+Given:
+
+There is a package `prop-types` with a default export
+
+and
+
+```json
+// .eslintrc
+{
+  "rules": {
+    "import/rename-default-import": [
+      "warn", {
+        "prop-types": "PropTypes", // key: name of the module, value: desired binding for default import
```

Comment:
what validates that the RHS of this object is a valid JS identifier?

Reply:
At the moment it is only the user of the plugin who should be aware what they are doing. I could do something like ``eval(`(() => {var ${mapping}})()`)`` or ``new Function(`var ${mapping}`)``

Reply:
Even though it is not the safest way to check for this..

Reply:
is checking that even important? sounds like a case of garbage-in, garbage-out if people put invalid stuff there...

Reply:
I prefer my projects not to ever emit garbage :-)

Reply:
If we decide to validate the user-provided identifier I would like to find out an easy way to do it. Because so far I couldn't find a simple and secure solution.

Reply:
Hi @mic4ael
Sorry for reincarnating this thread after such a long time. My take on this would be to parse either `var ${mapping}` or `${mapping}` using `espree` - a module also used internally by `eslint` for parsing code.

In case of `${mapping}`:
https://astexplorer.net/#/gist/94e57078a7e95dbf910a262c645dd3ea/b82ba422cd8e946bee02a1209795a1bbaa9be225

The parsed AST should consist of exactly one `ExpressionStatement` having an `Identifier` set as its `expression`. Everything else (or an exception)  is treated as an incorrect `mapping`.

In case of `var ${mapping}`:
https://astexplorer.net/#/gist/40823431b92aaf01426b255e4ad7163f/24a943d0b759cdf8aff897dfa1c5dcba48cb3847

The parsed AST should consist of exactly one `VariableDeclaration`, which has its `init` set to `null` and its `id` set to an `Identifier`.

That way you can be sure `mapping` is no garbage.